lib/stanford-mods/searchworks_subjects.rb

```diff
@@ -0,0 +1,208 @@
+      def geographic_search
+        @geographic_search ||= begin
+          result = self.sw_geographic_search
+
+          # TODO:  this should go into stanford-mods ... but then we have to set that gem up with a Logger
+          # print a message for any unrecognized encodings
+          xvals = self.subject.geographicCode.translated_value
+          codes = self.term_values([:subject, :geographicCode])
+          if codes && codes.size > xvals.size
+            self.subject.geographicCode.each { |n|
+              if n.authority != 'marcgac' && n.authority != 'marccountry'
+                sw_logger.info("#{druid} has subject geographicCode element with untranslated encoding (#{n.authority}): #{n.to_xml}")
+              end
+            }
+          end
+
+          # FIXME:  stanford-mods should be returning [], not nil ...
+          return nil if !result || result.empty?
+          result
+        end
+      end
+
+      # Values are the contents of:
+      #   subject/name
+      #   subject/occupation  - no subelements
+      #   subject/titleInfo
+      # @return [Array<String>] values for the subject_other_search Solr field for this document or nil if none
+      def subject_other_search
+        @subject_other_search ||= begin
+          vals = subject_occupations ? Array.new(subject_occupations) : []
+          vals.concat(subject_names) if subject_names
+          vals.concat(subject_titles) if subject_titles
+          vals.empty? ? nil : vals
+        end
+      end
+
+      # Values are the contents of:
+      #   subject/temporal
+      #   subject/genre
+      # @return [Array<String>] values for the subject_other_subvy_search Solr field for this document or nil if none
+      def subject_other_subvy_search
+        @subject_other_subvy_search ||= begin
+          vals = subject_temporal ? Array.new(subject_temporal) : []
+          gvals = self.term_values([:subject, :genre])
+          vals.concat(gvals) if gvals
+
+          # print a message for any temporal encodings
+          self.subject.temporal.each { |n|
+            sw_logger.info("#{druid} has subject temporal element with untranslated encoding: #{n.to_xml}") if !n.encoding.empty?
+          }
+
+          vals.empty? ? nil : vals
+        end
+      end
+
+      # Values are the contents of:
+      #  all subject subelements except subject/cartographic plus  genre top level element
+      # @return [Array<String>] values for the subject_all_search Solr field for this document or nil if none
+      def subject_all_search
+        vals = topic_search ? Array.new(topic_search) : []
+        vals.concat(geographic_search) if geographic_search
+        vals.concat(subject_other_search) if subject_other_search
+        vals.concat(subject_other_subvy_search) if subject_other_subvy_search
+        vals.empty? ? nil : vals
+      end
+
+# protected ----------------------------------------------------------
```

Comment:
I know you're just copying code around, but do you know why we say these are protected, but haven't actually made it protected?